src/core/common/settings.cpp

```diff
@@ -0,0 +1,324 @@
+
+    otLogInfoCore(GetInstance(), "Non-volatile: %s ChildInfo {rloc:0x%04x, extaddr:%s, timeout:%u, mode:0x%02x}",
+                  aAction, aChildInfo.mRloc16, aChildInfo.mExtAddress.ToString(string, sizeof(string)),
+                  aChildInfo.mTimeout, aChildInfo.mMode);
+}
+
+#endif // #if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_INFO)
+
+#if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_WARN)
+
+void SettingsBase::LogFailure(otError error, const char *aText) const
+{
+    if (error != OT_ERROR_NONE)
+    {
+        otLogWarnCore(GetInstance(), "Non-volatile: Error %s %s", otThreadErrorToString(error), aText);
+    }
+}
+
+#endif // #if (OPENTHREAD_CONFIG_LOG_LEVEL >= OT_LOG_LEVEL_WARN)
+#endif // #if (OPENTHREAD_CONFIG_LOG_UTIL != 0)
+
+void Settings::Init(void)
+{
+    otPlatSettingsInit(&GetInstance());
+}
+
+void Settings::Wipe(void)
+{
+    otPlatSettingsWipe(&GetInstance());
+    otLogInfoCore(GetInstance(), "Non-volatile: Wiped all info");
+}
+
+otError Settings::SaveOperationalDataset(bool aIsActive, const MeshCoP::Dataset &aDataset)
+{
+    otError error = Save(aIsActive ? kKeyActiveDataset : kKeyPendingDataset, aDataset.GetBytes(), aDataset.GetSize());
+
+    LogFailure(error, "saving OperationalDataset");
+    return error;
+}
+
+otError Settings::ReadOperationalDataset(bool aIsActive, MeshCoP::Dataset &aDataset) const
+{
+    otError  error = OT_ERROR_NONE;
+    uint16_t length;
+
+    SuccessOrExit(error = Read(aIsActive ? kKeyActiveDataset : kKeyPendingDataset, aDataset.GetBytes(),
+                               MeshCoP::Dataset::kMaxSize, length));
+    aDataset.SetSize(length);
+
+exit:
+    return error;
+}
+
+otError Settings::DeleteOperationalDataset(bool aIsActive)
+{
+    otError error = Delete(aIsActive ? kKeyActiveDataset : kKeyPendingDataset);
+
+    LogFailure(error, "deleting OperationalDataset");
+    return error;
+}
+
+otError Settings::ReadNetworkInfo(NetworkInfo &aNetworkInfo) const
+{
+    otError error;
+
+    SuccessOrExit(error = ReadFixedSize(kKeyNetworkInfo, &aNetworkInfo, sizeof(NetworkInfo)));
+    LogNetworkInfo("Read", aNetworkInfo);
+
+exit:
+    return error;
+}
+
+otError Settings::SaveNetworkInfo(const NetworkInfo &aNetworkInfo)
+{
+    otError error;
+
+    SuccessOrExit(error = Save(kKeyNetworkInfo, &aNetworkInfo, sizeof(NetworkInfo)));
+    LogNetworkInfo("Saved", aNetworkInfo);
+
+exit:
+    LogFailure(error, "saving NetworkInfo");
+    return error;
+}
+
+otError Settings::DeleteNetworkInfo(void)
+{
+    otError error;
+
+    SuccessOrExit(error = Delete(kKeyNetworkInfo));
+    otLogInfoCore(GetInstance(), "Non-volatile: Deleted NetworkInfo");
+
+exit:
+    LogFailure(error, "deleting NetworkInfo");
+    return error;
+}
+
+otError Settings::ReadParentInfo(ParentInfo &aParentInfo) const
+{
+    otError error;
+
+    SuccessOrExit(error = ReadFixedSize(kKeyParentInfo, &aParentInfo, sizeof(ParentInfo)));
+    LogParentInfo("Read", aParentInfo);
+
+exit:
+    return error;
```

Comment:
Add `LogFailure`?

Reply:
Internationally did not include it on all "read" methods. My thought was that  on "read" if there is no value in settings store we can get an `OT_ERROR_NOT_FOUND` which I felt is not an error specially at warning log level. 

I am ok adding `LogFailure` here. @jwhui  what do you think?

Reply:
Makes sense not to include the log.  Thanks.